splinepy/microstructure/tiles/tile_base.py

```diff
@@ -8,18 +8,33 @@ class TileBase(_SplinepyBase):
     Base class for tile objects
     """
 
+    _dim = None
+    _para_dim = None
+    _evaluation_points = None
+    _n_info_per_eval_point = None
+
     def __init__(self):
-        """
-        Init Values to None
-        """
-        self._dim = None
-        self._evaluation_points = None
-        self._n_info_per_eval_point = None
+        if type(self) is TileBase:
+            raise TypeError(
+                "Can't initialize a TileBase class! Please use one of the "
+                f"derived classes: {TileBase.__subclasses__()}"
+            )
```

Comment:
I assume this code is supposed to prevent an instantiation of an instance of type `TileBase`. Is there a reason why you decided to go this way instead of inheriting from `abc.ABC`?

Reply:
Yes, that would be good and I think we used to have it at the very beginning. I had some cases where abc made things a bit more complicated, so I decided to stay with simple inheritance

Reply:
Okay, thanks for clarifying! 👍